### PR TITLE
[MOBILE-2140] Add HMS module to Cordova

### PR DESCRIPTION
**Pull request by Apekka**
### What do these changes do?
Adding the HMS module to Cordova

### Why are these changes necessary?
So Cordova apps can support HMS devices

### How did you verify these changes?
By adding the plugin to the sample app and verifying that it was correctly implemented

### CODE REVIEW

File: urbanairship-hms-cordova/package.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  "cordova": {
+    "id": "urbanairship-hms-cordova",
+    "platforms": [
+      "ios",
+      "android"
+    ]
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/urbanairship/urbanairship-hms-cordova.git"
```
Comment (marc-scig) [inline urbanairship-hms-cordova/package.json:14]: these urls are pointing to a non-existent repo

File: urbanairship-hms-cordova/package.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      "version": ">=5.0.1"
+    },
+    {
+      "name": "cordova-plugman",
+      "version": ">=4.2.0"
+    }
+  ],
+  "author": "Urban Airship",
+  "license": "Apache 2.0",
+  "homepage": "https://github.com/urbanairship/urbanairship-hms-cordova.git",
```
Comment (marc-scig) [inline urbanairship-hms-cordova/package.json:39]: we should probably point these to this repo instead

File: urbanairship-hms-cordova/plugin.xml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+<plugin id="urbanairship-hms-cordova"
+        version="12.2.0"
+        xmlns="http://apache.org/cordova/ns/plugins/1.0"
+        xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <name>urbanairship-hms-cordova</name>
+    <description>Urban Airship HMS Cordova plugin</description>
+    <license>Apache 2.0</license>
+    <keywords>cordova,urbanairship</keywords>
+    <repo>https://github.com/urbanairship/urbanairship-hms-cordova.git</repo>
```
Comment (marc-scig) [inline urbanairship-hms-cordova/plugin.xml:11]: https://github.com/urbanairship/urbanairship-cordova.git

File: urbanairship-hms-cordova/package.json [hunk 4]
```diff
@@ -0,0 +1,43 @@
+{
+  "name": "urbanairship-hms-cordova",
+  "version": "12.1.1",
+  "description": "Urban Airship HMS Cordova plugin",
+  "cordova": {
+    "id": "urbanairship-hms-cordova",
+    "platforms": [
+      "ios",
```
Comment (marc-scig) [inline urbanairship-hms-cordova/package.json:8]: this plugin is just for android, right?

**Review by marc-scig — APPROVED** [review]
+1 with a minor change

File: urbanairship-hms-cordova/src/android/build-extras.gradle [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+allprojects {
+    repositories {
+        google()
+        jcenter()
+    }
+}
+
+dependencies {
+    implementation 'com.urbanairship.android:urbanairship-hms:14.3.0'
```
Comment (rlepinski) [inline urbanairship-hms-cordova/src/android/build-extras.gradle:21]: probably add "com.huawei.hms:push:$rootProject.huaweiPushKitVersion"

File: urbanairship-hms-cordova/src/android/build-extras.gradle [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+// For CI only. Verify our plugin is Java 6 compatible
+if (project.hasProperty('uaInternalJava6CompileOptions') && uaInternalJava6CompileOptions.toBoolean()) {
+    cdvPluginPostBuildExtras.push({
+        android.compileOptions {
+            sourceCompatibility JavaVersion.VERSION_1_6
+            targetCompatibility JavaVersion.VERSION_1_6
+        }
+    })
+}
```
Comment (rlepinski) [inline urbanairship-hms-cordova/src/android/build-extras.gradle:34]: At the end `apply plugin: 'com.huawei.agconnect'

File: urbanairship-hms-cordova/src/android/build-extras.gradle [hunk 7]
```diff
@@ -0,0 +1,34 @@
+
+buildscript {
+    repositories {
+        google()
+        jcenter()
```
Comment (rlepinski) [inline urbanairship-hms-cordova/src/android/build-extras.gradle:5]: maven { url 'http://developer.huawei.com/repo/' }

File: urbanairship-hms-cordova/src/android/build-extras.gradle [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    dependencies {
+        classpath 'com.android.tools.build:gradle:3.5.3'
```
Comment (rlepinski) [inline urbanairship-hms-cordova/src/android/build-extras.gradle:9]: classpath "com.huawei.agconnect:agcp:$huaweiGradlePluginVersion"

**Comment by rlepinski** [comment]
Not sure what the versions are for HMS but Goat is using:
huaweiPushKitVersion = "5.0.4.302"
huaweiGradlePluginVersion = "1.4.2.301"

File: urbanairship-hms-cordova/package.json [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    "huawei",
+    "cordova-android"
+  ],
+  "engines": [
+    {
+      "name": "cordova-android",
+      "version": ">=4.1.0"
+    },
+    {
+      "name": "cordova-ios",
```
Comment (Ulrico972) [inline urbanairship-hms-cordova/package.json:28]: I think you can remove this

File: urbanairship-hms-cordova/README.md [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+This plugin supports Cordova apps running on both iOS and Android.
+
+### Issues
+
+Please visit http://support.urbanairship.com/ for any issues integrating or using this plugin.
+
+### Requirements:
+ - cordova >= 9.0.1
+ - cordova-ios >= 5.0.1
```
Comment (Ulrico972) [inline urbanairship-hms-cordova/README.md:11]: You should remove the cordova-ios and cocoapods requirements.
Same for the Xcode mention below

**Review by rlepinski — APPROVED** [review]

**Pull request merged**